src/Monitoring/DotNet/Monitoring.EventProcessor/EventHubMonitor.cs

```diff
@@ -0,0 +1,124 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Reactive.Concurrency;
+using System.Reactive.Linq;
+using System.Reactive.Subjects;
+using System.Reactive.Threading.Tasks;
+using System.Threading.Tasks;
+using Microsoft.ServiceBus.Messaging;
+
+namespace Microsoft.Practices.IoTJourney.Monitoring.EventProcessor
+{
+    public class EventHubMonitor : IObservable<PartitionSnapshot>
```

Comment:
Thanks @LeeCampbell. Following that pattern then, what work would you do in the delegate for `Observable.Create`? Would it be what I'm currently doing the `GenerateStream` method? I thought that using a subject as an intermediary was another recommended way to avoid implementing `IObservable<T>`? 

I could make a few semantic changes and turn the `EventHubMonitorFactory` into the object that has-as instance of the stream and remove the current `EventHubMonitor` altogether. That could mean that all of the current logic in `EventHubMonitor` would move into the delegate passed into `Observable.Create`. Is that more inline with what you're thinking?